essentials-framework/Essentials Core/PepperDashEssentialsBase/File/FileIO.cs

```diff
@@ -0,0 +1,163 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using Crestron.SimplSharp;
+using Crestron.SimplSharp.CrestronIO;
+using PepperDash.Core;
+
+namespace PepperDash.Essentials.Core
+{
+	public static class FileIO
+	{
+
+		static CCriticalSection fileLock = new CCriticalSection();
+		/// <summary>
+		/// 
+		/// </summary>
+		/// <param name="fileName"></param>
+		/// <returns></returns>
+		public static bool FileExsists(string fileName)
+		{
+			if (GetFile(fileName) == null)
+			{
+				return false;
+			}
+			else
+			{
+				return true;
+			}
+		}
+		/// <summary>
+		/// Get the full file info from a path/filename, can include wildcards.
+		/// </summary>
+		/// <param name="fileName"></param>
+		/// <returns></returns>
+		public static FileInfo[] GetFiles(string fileName)
+		{
+			DirectoryInfo dirInfo = new DirectoryInfo(Path.GetDirectoryName(fileName));
+			var files = dirInfo.GetFiles(Path.GetFileName(fileName));
+			Debug.Console(0, "FileIO found: {0}, {1}", files.Count(), fileName);
+			if (files.Count() > 0)
+			{
+				return files;
+			}
+			else
+			{
+				return null;
+			}
+		}
+
+		public static FileInfo GetFile(string fileName)
+		{
+			DirectoryInfo dirInfo = new DirectoryInfo(Path.GetDirectoryName(fileName));
+			var files = dirInfo.GetFiles(Path.GetFileName(fileName));
+			Debug.Console(0, "FileIO found: {0}, {1}", files.Count(), fileName);
+			if (files.Count() > 0)
+			{
+				return files.FirstOrDefault();
+			}
+			else
+			{
+				return null;
+			}
+		}
+	
+
+		/// <summary>
+		/// Get the data from a fileName
+		/// </summary>
+		/// <param name="fileName"></param>
+		/// <returns></returns>
+		public static string GetDataFromFile(FileInfo file)
```

Comment:
We could possibly consider renaming this `ReadDataFromFile` to better match with the paired `WriteDataToFile` method?

Reply:
Should we also consider an overload of this method that just takes the filename as a string and returns the data.  Would preclude needing to call `GetFile` to get the file info first if not necessary?